File: sdkwork-ai-sdk-go/api/moderation.go
package api

import (
    sdktypes "github.com/sdkwork/ai-sdk/types"
    sdkhttp "github.com/sdkwork/ai-sdk/http"
)

type ModerationApi struct {
    client *sdkhttp.Client
}

func NewModerationApi(client *sdkhttp.Client) *ModerationApi {
    return &ModerationApi{client: client}
}

// Create moderation
func (a *ModerationApi) CreateModeration(body sdktypes.ModerationRequest) (sdktypes.ModerationResponse, error) {
    raw, err := a.client.Post(AiApiPath("/moderations"), body, nil, nil, "")
    if err != nil {
        var zero sdktypes.ModerationResponse
        return zero, err
    }
    return decodeResult[sdktypes.ModerationResponse](raw)
}

File: sdkwork-ai-sdk-typescript/src/api/moderation.ts
import { aiApiPath } from './paths';
import type { HttpClient } from '../http/client';
import type { QueryParams } from '../types/common';
import type { ModerationRequest, ModerationResponse } from '../types';


export class ModerationApi {
  private client: HttpClient;
  
  constructor(client: HttpClient) { 
    this.client = client; 
  }

/** Create moderation */
  async createModeration(body: ModerationRequest): Promise<ModerationResponse> {
    return this.client.post<ModerationResponse>(aiApiPath(`/moderations`), body);
  }
}

export function createModerationApi(client: HttpClient): ModerationApi {
  return new ModerationApi(client);
}
